[BUG] Kyverno panics on interface conversion

**Software version numbers**
State the version numbers of applications involved in the bug.

* Kubernetes version: 1.21.3
* Kubernetes platform (if applicable; ex., EKS, GKE, OpenShift): K3d
* Kyverno version: 1.5.0

**Describe the bug**
Kyverno panics with an interface conversion error.

**To Reproduce**
Steps to reproduce the behavior:
1. Apply this WIP policy.

```yaml
apiVersion: kyverno.io/v1
kind: ClusterPolicy
metadata:
  name: limit-hostpath-vols
  annotations:
    policies.kyverno.io/title: Limit hostPath Volumes to Directories
    policies.kyverno.io/category: Other
    policies.kyverno.io/severity: medium
    policies.kyverno.io/minversion: 1.5.0
    policies.kyverno.io/description: >-
      Pods which are allowed to mount hostPath volumes in read/write mode pose a security risk
      even if confined to a "safe" file system on the host and may escape those confines (see
      https://blog.aquasec.com/kubernetes-security-pod-escape-log-mounts). The only true way
      to ensure safety is to enforce that all Pods mounting hostPath volumes do so in read only
      mode. This policy checks all containers for any hostPath volumes and ensures they are
      explicitly mounted in readOnly mode.
spec:
  background: false
  validationFailureAction: audit
  rules:
  - name: limit-hostpath-to-slash-data
    match:
      resources:
        kinds:
        - Pod
    preconditions:
      all:
      - key: "{{ request.object.spec.volumes[?hostPath] | length(@) }}"
        operator: GreaterThanOrEquals
        value: 1
      - key: "{{request.operation}}"
        operator: In
        value:
        - CREATE
        - UPDATE
    validate:
      message: hostPath volumes may only mount /data.
      deny:
        conditions:
        - key: "{{ request.object.spec.volumes[?hostPath].hostPath.path.split(@, '/') | [1] | contains(@, 'data') }}"
          operator: Equals
          value: false
```
2. Create this sample resource.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: bad-pods-all
spec:
  ephemeralContainers:
  - name: ephemtest-webserver
    image: asdfeasdfasada:latest
    volumeMounts:
    - mountPath: /some/dir
      name: foo
      readOnly: false
  initContainers:
  - name: inittest-webserver
    image: asdfeasdfasada:latest
    volumeMounts:
    - mountPath: /some/dir
      name: bar
      readOnly: true
  containers:
  - name: test-webserver
    image: asdfeasdfasada:latest
    volumeMounts:
    - mountPath: /some/dir
      name: foo
      readOnly: true
  - name: test-webserver02
    image: sjbonmqopcta:latest
    volumeMounts:
    - mountPath: /some/dir
      name: bar
      readOnly: true
  volumes:
  - name: foo
    hostPath:
      path: /var/log
  - name: bar
    hostPath:
      path: /etc
  - name: empty
    emptyDir:
      medium: memory
      sizeLimit: 20Mi
```
3. See the panic in the logs:

```log
E1025 20:39:03.629945       1 runtime.go:78] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x1df0420), concrete:(*runtime._type)(0x1d16f00), asserted:(*runtime._type)(0x1d0dc00), missingMethod:""} (interface conversion: interface {} is []string, not []interface {})
goroutine 1469 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1e63e40, 0xc003643800)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/runtime/runtime.go:48 +0x86
panic(0x1e63e40, 0xc003643800)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/jmespath/go-jmespath.jpfContains(0xc002bc2960, 0x2, 0x2, 0x2, 0xc002bc2960, 0x2, 0x2)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/functions.go:453 +0x225
github.com/jmespath/go-jmespath.(*functionCaller).CallFunction(0xc0025b8268, 0xc001c3204c, 0x8, 0xc002bc2960, 0x2, 0x2, 0xc0025b8258, 0x2, 0xc003fc5638, 0x1d5f560, ...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/functions.go:401 +0x197
github.com/jmespath/go-jmespath.(*treeInterpreter).Execute(0xc0025b8258, 0x4, 0x1d5f560, 0xc003b180e0, 0xc001375080, 0x2, 0x2, 0x1d16f00, 0xc003fc5638, 0x1d16f00, ...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/interpreter.go:93 +0x1ed2
github.com/jmespath/go-jmespath.(*treeInterpreter).Execute(0xc0025b8258, 0x12, 0x0, 0x0, 0xc001375260, 0x2, 0x2, 0x1e25a00, 0xc00369fec0, 0x0, ...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/interpreter.go:276 +0xbaf
github.com/jmespath/go-jmespath.(*JMESPath).Search(...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/api.go:37
github.com/kyverno/kyverno/pkg/engine/context.(*Context).Query(0xc000c55880, 0xc001c32001, 0x5e, 0x0, 0x0, 0x0, 0x0)
        /kyverno/pkg/engine/context/evaluate.go:46 +0x345
github.com/kyverno/kyverno/pkg/engine/variables.DefaultVariableResolver(0x2974500, 0xc000c55880, 0xc001c32001, 0x5e, 0xc000ad37c0, 0x1, 0xa, 0x0)
        /kyverno/pkg/engine/variables/vars.go:327 +0x45
github.com/kyverno/kyverno/pkg/engine/variables.substituteVariablesIfAny.func1(0xc003083590, 0x40e398, 0x30, 0x1f44f20, 0x1)
        /kyverno/pkg/engine/variables/vars.go:354 +0x1c5
github.com/kyverno/kyverno/pkg/engine/json-utils.OnlyForLeafsAndKeys.func1(0xc003083590, 0xc003083590, 0xc001550253, 0xc00069c460, 0x0)
        /kyverno/pkg/engine/json-utils/traverse.go:31 +0x79
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseJSON(0xc00069c700, 0x1d5f560, 0xc0039a73a0, 0xc001550250, 0x6, 0xc000898cfc, 0x3, 0xc001550250, 0x6)
        /kyverno/pkg/engine/json-utils/traverse.go:64 +0xb0
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseObject(0xc00069c700, 0xc0030b7a70, 0xc000ba8c40, 0x2, 0x0, 0x10, 0xc00069c5b0)
        /kyverno/pkg/engine/json-utils/traverse.go:104 +0x205
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseJSON(0xc00069c700, 0x1e25a00, 0xc0030b7890, 0xc000ba8c40, 0x2, 0x2152579, 0x1, 0xc000ba8c40, 0x2)
        /kyverno/pkg/engine/json-utils/traverse.go:72 +0x159
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseList(0xc00069c700, 0xc0039a7430, 0x1, 0x1, 0x0, 0x0, 0xc00069c688, 0x1b0182d, 0x1ed1a40, 0xc0039a7420, ...)
        /kyverno/pkg/engine/json-utils/traverse.go:123 +0xfd
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseJSON(0xc00069c700, 0x1d0dc00, 0xc003fc49d8, 0x0, 0x0, 0xc0039a7420, 0xc003fc49d8, 0x0, 0x0)
        /kyverno/pkg/engine/json-utils/traverse.go:75 +0x2aa
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).TraverseJSON(...)
        /kyverno/pkg/engine/json-utils/traverse.go:59
github.com/kyverno/kyverno/pkg/engine/variables.substituteVars(0x29b6fe8, 0xc001db2f00, 0x2974500, 0xc000c55880, 0x1d0dc00, 0xc003fc49d8, 0x279d3d8, 0x0, 0xc001db2f00, 0x2974500, ...)
        /kyverno/pkg/engine/variables/vars.go:192 +0xb7
github.com/kyverno/kyverno/pkg/engine/variables.substituteAll(0x29b6fe8, 0xc001db2f00, 0x2974500, 0xc000c55880, 0x1d0dc00, 0xc0037efb30, 0x279d3d8, 0x0, 0xc003955aa0, 0x2, ...)
        /kyverno/pkg/engine/variables/vars.go:163 +0xdf
github.com/kyverno/kyverno/pkg/engine/variables.SubstituteAll(...)
        /kyverno/pkg/engine/variables/vars.go:63
github.com/kyverno/kyverno/pkg/engine.(*validator).validateDeny(0xc000314540, 0xc001db2f00)
        /kyverno/pkg/engine/validation.go:305 +0x89
github.com/kyverno/kyverno/pkg/engine.(*validator).validate(0xc000314540, 0xc001db2f00)
        /kyverno/pkg/engine/validation.go:200 +0x156
github.com/kyverno/kyverno/pkg/engine.processValidationRule(0x29b6fe8, 0xc001db2f00, 0xc000d21b00, 0xc001c8b880, 0x0)
        /kyverno/pkg/engine/validation.go:118 +0x88
github.com/kyverno/kyverno/pkg/engine.validateResource(0x29b6fe8, 0xc001db2f00, 0xc000d21b00, 0x0)
        /kyverno/pkg/engine/validation.go:103 +0x505
github.com/kyverno/kyverno/pkg/engine.Validate(0xc000d21b00, 0xc000138600)
        /kyverno/pkg/engine/validation.go:37 +0x252
github.com/kyverno/kyverno/pkg/webhooks.(*validationHandler).handleValidation(0xc00069dd28, 0xc00059b980, 0xc0011f7520, 0xc00149c950, 0x1, 0x1, 0xc000d21b00, 0xc0006fea50, 0x617715e7, 0x0, ...)
        /kyverno/pkg/webhooks/validation.go:64 +0x5cb
github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).process(0xc0001db1e0, 0xc0011f7520, 0xc0011f7520, 0x415200)
        /kyverno/pkg/webhooks/validate_audit.go:208 +0x605
github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).processNextWorkItem(0xc0001db1e0, 0x203000)
        /kyverno/pkg/webhooks/validate_audit.go:144 +0x1d8
github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).runWorker(...)
        /kyverno/pkg/webhooks/validate_audit.go:125
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0018b0050)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0018b0050, 0x29672c0, 0xc0008d3d40, 0x1, 0xc0000ac3c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0018b0050, 0x3b9aca00, 0x0, 0xc000056b01, 0xc0000ac3c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc0018b0050, 0x3b9aca00, 0xc0000ac3c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:90 +0x4d
created by github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).Run
        /kyverno/pkg/webhooks/validate_audit.go:118 +0x132
panic: interface conversion: interface {} is []string, not []interface {} [recovered]
        panic: interface conversion: interface {} is []string, not []interface {}

goroutine 1469 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/runtime/runtime.go:55 +0x109
panic(0x1e63e40, 0xc003643800)
        /usr/local/go/src/runtime/panic.go:965 +0x1b9
github.com/jmespath/go-jmespath.jpfContains(0xc002bc2960, 0x2, 0x2, 0x2, 0xc002bc2960, 0x2, 0x2)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/functions.go:453 +0x225
github.com/jmespath/go-jmespath.(*functionCaller).CallFunction(0xc0025b8268, 0xc001c3204c, 0x8, 0xc002bc2960, 0x2, 0x2, 0xc0025b8258, 0x2, 0xc003fc5638, 0x1d5f560, ...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/functions.go:401 +0x197
github.com/jmespath/go-jmespath.(*treeInterpreter).Execute(0xc0025b8258, 0x4, 0x1d5f560, 0xc003b180e0, 0xc001375080, 0x2, 0x2, 0x1d16f00, 0xc003fc5638, 0x1d16f00, ...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/interpreter.go:93 +0x1ed2
github.com/jmespath/go-jmespath.(*treeInterpreter).Execute(0xc0025b8258, 0x12, 0x0, 0x0, 0xc001375260, 0x2, 0x2, 0x1e25a00, 0xc00369fec0, 0x0, ...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/interpreter.go:276 +0xbaf
github.com/jmespath/go-jmespath.(*JMESPath).Search(...)
        /go/pkg/mod/github.com/kyverno/go-jmespath@v0.4.1-0.20210511164400-a1d46efa2ed6/api.go:37
github.com/kyverno/kyverno/pkg/engine/context.(*Context).Query(0xc000c55880, 0xc001c32001, 0x5e, 0x0, 0x0, 0x0, 0x0)
        /kyverno/pkg/engine/context/evaluate.go:46 +0x345
github.com/kyverno/kyverno/pkg/engine/variables.DefaultVariableResolver(0x2974500, 0xc000c55880, 0xc001c32001, 0x5e, 0xc000ad37c0, 0x1, 0xa, 0x0)
        /kyverno/pkg/engine/variables/vars.go:327 +0x45
github.com/kyverno/kyverno/pkg/engine/variables.substituteVariablesIfAny.func1(0xc003083590, 0x40e398, 0x30, 0x1f44f20, 0x1)
        /kyverno/pkg/engine/variables/vars.go:354 +0x1c5
github.com/kyverno/kyverno/pkg/engine/json-utils.OnlyForLeafsAndKeys.func1(0xc003083590, 0xc003083590, 0xc001550253, 0xc00069c460, 0x0)
        /kyverno/pkg/engine/json-utils/traverse.go:31 +0x79
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseJSON(0xc00069c700, 0x1d5f560, 0xc0039a73a0, 0xc001550250, 0x6, 0xc000898cfc, 0x3, 0xc001550250, 0x6)
        /kyverno/pkg/engine/json-utils/traverse.go:64 +0xb0
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseObject(0xc00069c700, 0xc0030b7a70, 0xc000ba8c40, 0x2, 0x0, 0x10, 0xc00069c5b0)
        /kyverno/pkg/engine/json-utils/traverse.go:104 +0x205
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseJSON(0xc00069c700, 0x1e25a00, 0xc0030b7890, 0xc000ba8c40, 0x2, 0x2152579, 0x1, 0xc000ba8c40, 0x2)
        /kyverno/pkg/engine/json-utils/traverse.go:72 +0x159
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseList(0xc00069c700, 0xc0039a7430, 0x1, 0x1, 0x0, 0x0, 0xc00069c688, 0x1b0182d, 0x1ed1a40, 0xc0039a7420, ...)
        /kyverno/pkg/engine/json-utils/traverse.go:123 +0xfd
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).traverseJSON(0xc00069c700, 0x1d0dc00, 0xc003fc49d8, 0x0, 0x0, 0xc0039a7420, 0xc003fc49d8, 0x0, 0x0)
        /kyverno/pkg/engine/json-utils/traverse.go:75 +0x2aa
github.com/kyverno/kyverno/pkg/engine/json-utils.(*Traversal).TraverseJSON(...)
        /kyverno/pkg/engine/json-utils/traverse.go:59
github.com/kyverno/kyverno/pkg/engine/variables.substituteVars(0x29b6fe8, 0xc001db2f00, 0x2974500, 0xc000c55880, 0x1d0dc00, 0xc003fc49d8, 0x279d3d8, 0x0, 0xc001db2f00, 0x2974500, ...)
        /kyverno/pkg/engine/variables/vars.go:192 +0xb7
github.com/kyverno/kyverno/pkg/engine/variables.substituteAll(0x29b6fe8, 0xc001db2f00, 0x2974500, 0xc000c55880, 0x1d0dc00, 0xc0037efb30, 0x279d3d8, 0x0, 0xc003955aa0, 0x2, ...)
        /kyverno/pkg/engine/variables/vars.go:163 +0xdf
github.com/kyverno/kyverno/pkg/engine/variables.SubstituteAll(...)
        /kyverno/pkg/engine/variables/vars.go:63
github.com/kyverno/kyverno/pkg/engine.(*validator).validateDeny(0xc000314540, 0xc001db2f00)
        /kyverno/pkg/engine/validation.go:305 +0x89
github.com/kyverno/kyverno/pkg/engine.(*validator).validate(0xc000314540, 0xc001db2f00)
        /kyverno/pkg/engine/validation.go:200 +0x156
github.com/kyverno/kyverno/pkg/engine.processValidationRule(0x29b6fe8, 0xc001db2f00, 0xc000d21b00, 0xc001c8b880, 0x0)
        /kyverno/pkg/engine/validation.go:118 +0x88
github.com/kyverno/kyverno/pkg/engine.validateResource(0x29b6fe8, 0xc001db2f00, 0xc000d21b00, 0x0)
        /kyverno/pkg/engine/validation.go:103 +0x505
github.com/kyverno/kyverno/pkg/engine.Validate(0xc000d21b00, 0xc000138600)
        /kyverno/pkg/engine/validation.go:37 +0x252
github.com/kyverno/kyverno/pkg/webhooks.(*validationHandler).handleValidation(0xc00069dd28, 0xc00059b980, 0xc0011f7520, 0xc00149c950, 0x1, 0x1, 0xc000d21b00, 0xc0006fea50, 0x617715e7, 0x0, ...)
        /kyverno/pkg/webhooks/validation.go:64 +0x5cb
github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).process(0xc0001db1e0, 0xc0011f7520, 0xc0011f7520, 0x415200)
        /kyverno/pkg/webhooks/validate_audit.go:208 +0x605
github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).processNextWorkItem(0xc0001db1e0, 0x203000)
        /kyverno/pkg/webhooks/validate_audit.go:144 +0x1d8
github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).runWorker(...)
        /kyverno/pkg/webhooks/validate_audit.go:125
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc0018b0050)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc0018b0050, 0x29672c0, 0xc0008d3d40, 0x1, 0xc0000ac3c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:156 +0x9b
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0018b0050, 0x3b9aca00, 0x0, 0xc000056b01, 0xc0000ac3c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc0018b0050, 0x3b9aca00, 0xc0000ac3c0)
        /go/pkg/mod/k8s.io/apimachinery@v0.21.4/pkg/util/wait/wait.go:90 +0x4d
created by github.com/kyverno/kyverno/pkg/webhooks.(*auditHandler).Run
        /kyverno/pkg/webhooks/validate_audit.go:118 +0x132
```

**Expected behavior**
Kyverno does not panic.

--- a/pkg/engine/jmespath/functions.go
+++ b/pkg/engine/jmespath/functions.go
@@ -336,7 +336,14 @@ func jpfSplit(arguments []interface{}) (interface{}, error) {
 		return nil, err
 	}
 
-	return strings.Split(str.String(), sep.String()), nil
+	split := strings.Split(str.String(), sep.String())
+	arr := make([]interface{}, len(split))
+
+	for i, v := range split {
+		arr[i] = v
+	}
+
+	return arr, nil
 }
 
 func jpRegexReplaceAll(arguments []interface{}) (interface{}, error) {

--- a/pkg/engine/jmespath/functions_test.go
+++ b/pkg/engine/jmespath/functions_test.go
@@ -128,7 +128,7 @@ func TestJMESPathFunctions_Split(t *testing.T) {
 	result, err := jp.Search("")
 	assert.NilError(t, err)
 
-	split, ok := result.([]string)
+	split, ok := result.([]interface{})
 	assert.Assert(t, ok)
 	assert.Equal(t, split[0], "Hello")
 	assert.Equal(t, split[1], "Gophers")

